warehouse/transformer/transformer.go

```diff
@@ -29,6 +29,14 @@ const (
 	redshiftStringLimit = 512
 )
 
+// Compile-time check to ensure ValidationError struct remains unchanged.
+var _ = struct {
+	Type     string
+	Message  string
+	Meta     map[string]string
+	Property string
+}(types.ValidationError{})
```

Comment:
why such compile-time check is necessary for this struct?

Reply:
Currently, we are relying on this struct to do the conversion from map[string]string to map[string]any. In case if something changes in future, we need to aware of it like they added new fields or modified some fields. If we don't do that it would break.
```GO
func convertToSliceIfViolationErrors(val any) any {
	if validationErrors, ok := val.([]types.ValidationError); ok {
		result := make([]any, len(validationErrors))
		for i, e := range validationErrors {
			result[i] = map[string]any{
				"type":     e.Type,
				"message":  e.Message,
				"meta":     lo.MapValues(e.Meta, func(value, _ string) any { return value }),
				"property": e.Property,
			}
		}
		return result
	}
	return val
}
```